crates/parse/src/parser/mod.rs

```diff
@@ -697,7 +697,8 @@ impl<'sess, 'ast> Parser<'sess, 'ast> {
         }
 
         if let Some(sep_kind) = &sep.sep {
-            if sep.trailing_sep_required && !trailing {
+            let open_close_delim = first && allow_empty;
```

Comment:
This just means we saw a pattern like `{}` and allows better error messages. Otherwise, the missing trailing comma distracts from the missing field error